stackslib/src/clarity_vm/database/mod.rs

```diff
@@ -316,16 +533,92 @@ where
     })
 }
 
-fn get_miner_column<F, R>(
+fn get_stacks_header_column<F, R>(
+    conn: &DBConn,
+    id_bhh: &StacksBlockId,
+    column_name: &str,
+    loader: F,
+) -> Option<R>
+where
+    F: Fn(&Row) -> R,
+{
+    match get_stacks_header_column_from_table(conn, id_bhh, column_name, &loader, false) {
+        Some(x) => Some(x),
+        None => get_stacks_header_column_from_table(conn, id_bhh, column_name, &loader, true),
+    }
+}
+
+fn get_first_block_in_tenure(
```

Comment:
Broadly speaking, this is just a placeholder.  When the chainstate is fully MARF'ed, the code here will instead simply query the headers MARF for the given Stacks block's first-tenure block (literally 3 lines of code).

Reply:
Ok cool. So then does this negate the need for the new issue you mentioned on line 593?

Reply:
No, please open the issue so I remember to do it after this merges :pray:

Reply:
Opened #4900